### PR TITLE
Update py-scikit-build-core to version 0.6.1

### PR DESCRIPTION
**Pull request by chrisrichardson**

### CODE REVIEW

**Review by wdconinc — CHANGES_REQUESTED** [review]
There are some changes to the requirements in the pyproject.toml at https://github.com/scikit-build/scikit-build-core/compare/v0.2.0...v0.6.0#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711

Relevant appears to be:
- `depends_on("py-importlib-metadata", when="@0.3.0: ^python@:3.7")`

**Comment by chrisrichardson** [comment]
I thought Python 3.7 was deprecated in Spack. Do we need to add support for it when adding new versions of other packages? (I can see the point when adding *old* versions)

**Comment by wdconinc** [comment]
Right you are. I keep mixing up the python versions supported in the package (https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/python/package.py) and the python versions supported by spack itself (https://spack.readthedocs.io/en/latest/getting_started.html). It's the former that's relevant here.

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the version sha256.

**Review by wdconinc — DISMISSED** [review]

**Comment by adamjstewart** [comment]
Personally I would still add the Python 3.7 dep since it hasn't been removed yet (I'll do that after SC). Also looks like a virtualenv dep is missing for testing.

**Comment by ma595** [comment]
@adamjstewart I have bumped to patch 0.6.1, could you please re-review?

File: var/spack/repos/builtin/packages/py-scikit-build-core/package.py [hunk 1]
```diff
@@ -49,6 +52,7 @@ class PyScikitBuildCore(PythonPackage):
     depends_on("py-pytest@7:", type="test")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-build-core/package.py:52]: Importlib-metadata no longer part of test in 0.5+

File: var/spack/repos/builtin/packages/py-scikit-build-core/package.py [hunk 2]
```diff
@@ -49,6 +52,7 @@ class PyScikitBuildCore(PythonPackage):
...
     depends_on("py-pytest-subprocess@1.5:", type="test")
     depends_on("py-setuptools", type="test")
+    depends_on("py-virtualenv", type="test")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-build-core/package.py]:
```suggestion
    depends_on("py-virtualenv", when="@0.6:", type="test")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adrianjhpc — APPROVED** [review]

**Review by adrianjhpc — APPROVED** [review]

**Review by adrianjhpc — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Review by wdconinc — APPROVED** [review]

**Comment by wdconinc** [comment]
I think in comparison with https://github.com/spack/spack/pull/41071, this PR is smaller (adds fewer versions), and already includes the virtualenv test requirement when 0.6:.

@adrianjhpc apologies, but looking forward to future PRs!

**Pull request merged**